packages/financial-templates-lib/src/price-feed/FundingRateMultiplierPriceFeed.ts

```diff
@@ -0,0 +1,144 @@
+import { PriceFeedInterface } from "./PriceFeedInterface";
+import { BlockFinder } from "./utils";
+import { ConvertDecimals } from "@uma/common";
+import { aggregateTransactionsAndCall } from "../helpers/multicall";
+import assert from "assert";
+import type { Logger } from "winston";
+import Web3 from "web3";
+import { PerpetualWeb3 } from "@uma/contracts-frontend";
+import { BN, Abi, Awaited } from "../types";
+import { BlockTransactionString } from "web3-eth";
+
+const _blockFinderGenericWorkaround = (requestBlock: (number: string | number) => Promise<BlockTransactionString>) => {
+  return BlockFinder(requestBlock);
+};
+
+type BlockFinder = ReturnType<typeof _blockFinderGenericWorkaround>;
+
+interface Params {
+  logger: Logger;
+  perpetualAbi: Abi;
+  web3: Web3;
+  perpetualAddress: string;
+  multicallAddress: string;
+  getTime: () => Promise<number>;
+  blockFinder?: BlockFinder;
+  minTimeBetweenUpdates?: number;
+  priceFeedDecimals?: number;
+}
+
+export class FundingRateMultiplierPriceFeed extends PriceFeedInterface {
+  private readonly logger: Logger;
+  private readonly web3: Web3;
+  private readonly getTime: () => Promise<number>;
+  private readonly multicallAddress: string;
+  private readonly blockFinder: BlockFinder;
+  private readonly minTimeBetweenUpdates: number;
+  private readonly priceFeedDecimals: number;
+  private lastUpdateTime: number | null = null;
+  private readonly uuid: string;
+  private price: BN | null = null;
+  private readonly convertDecimals: ReturnType<typeof ConvertDecimals>;
+  public readonly perpetual: PerpetualWeb3;
+
+  /**
+   * @notice Constructs new price feed object that tracks the funding rate multiplier in a particular perpetual contract.
+   * @param {Object} logger Winston module used to send logs.
+   * @param {Object} perpetualAbi Perpetual abi object to create a contract instance.
+   * @param {Object} web3 Provider from Truffle instance to connect to Ethereum network.
+   * @param {String} perpetualAddress Ethereum address of the perpetual to monitor.
+   * @param {String} multicallAddress Ethereum address of the multicall contract.
+   * @param {Function} getTime Returns the current time.
+   * @param {Function} [blockFinder] Optionally pass in a shared blockFinder instance (to share the cache).
+   * @param {Integer} [minTimeBetweenUpdates] Minimum time that must pass before update will actually run again.
+   * @param {Integer} [priceFeedDecimals] Precision that the caller wants precision to be reported in.
+   * @return None or throws an Error.
+   */
+  constructor({
+    logger,
+    perpetualAbi,
+    web3,
+    perpetualAddress,
+    multicallAddress,
+    getTime,
+    blockFinder,
+    minTimeBetweenUpdates = 60,
+    priceFeedDecimals = 18,
```

Comment:
you arent using the auto assign trick on these vars like you have in other classes

Reply:
It's a subtlety. Because they are passed in an object, you can't auto-assign (as far as I'm aware). If you're aware of some way to do that here, let me know!

Reply:
oh interesting ok!